src/features/chains/hooks.ts

```diff
@@ -28,3 +30,20 @@ export function useChainDisplayName(chainName: ChainName, shortName = false) {
   const multiProvider = useMultiProvider();
   return getChainDisplayName(multiProvider, chainName, shortName);
 }
+
+/**
+ * Return a chainMetadata object with given chains
+ */
+export function useChainMetadataMap(chains: string[]) {
+  const multiProvider = useMultiProvider();
+
+  const chainMetadata = useMemo(() => {
+    return chains.reduce<ChainMap<ChainMetadata>>((obj, chain) => {
+      const metadata = multiProvider.tryGetChainMetadata(chain);
+      if (metadata) obj[chain] = metadata;
+      return obj;
+    }, {});
+  }, [chains, multiProvider]);
+
+  return chainMetadata;
+}
```

Comment:
I don't think you need this hook. IIUC, `multiprovider.metadata` is already what you're looking for

Reply:
This hook is needed because we only need the token chains from the warp core, and that doesn't return the `metadata` we need, so we have to create it from scratch.

I renamed this function to `useTokenChainsMetadata` and also moved the `warpCore` and getting the chains logic inside